ADALiOS/ADALiOS/ADBrokerNotificationManager.m

```diff
@@ -68,20 +68,24 @@ -(void) enableOnActiveNotification:(ADAuthenticationCallback) callback
 
 - (void)callbackCleanup
 {
-    if(_callbackForBroker)
-    {
-        ADAuthenticationError* adError = [ADAuthenticationError errorFromAuthenticationError:AD_ERROR_BROKER_RESPONSE_NOT_RECEIVED
-                                                                                protocolCode:nil
-                                                                                errorDetails:@"application did not receive response from broker."];
-        ADAuthenticationResult* result = [ADAuthenticationResult resultFromError:adError];
-        ADAuthenticationCallback callback = _callbackForBroker;
-        callback(result);
-    }
-    
-    [[NSNotificationCenter defaultCenter] removeObserver:self
-                                                    name:UIApplicationDidBecomeActiveNotification
-                                                  object:nil];
-    _callbackForBroker = nil;
+    // We're not guaranteed the order that notifications happen in. Put this on the back of the main event queue so that
+    // launchURL might have a chance at the callback first.
```

Comment:
There's no particular way to guarantee always here. We're at Apple's mercy on the timing of various background/foreground notifications. They don't document the order of the foreground notifications compared to openURL and I've seen them occur in both orders. There isn't an even later foreground notification for us to take advantage of.

The hope is that by the point this notification gets hit, if the app was called via openUrl, that it's already in the queue (which seems to be the case). However the app might not have been called via URL, meaning we still need to make sure we still hit the completion block so the caller isn't left hanging. (Which would cause the app to get stuck.)